gendo/bot.py

```diff
@@ -31,10 +31,12 @@ def config_from_yaml(cls, path_to_yaml):
             settings = yaml.load(ymlfile)
             return cls(settings=settings)
 
-    def listen_for(self, rule, **options):
+    def listen_for(self, rule):
         def decorator(f):
-            self.add_listener(rule, f, **options)
-            return f
+            def wrapped(**kwargs):
+                self.add_listener(rule, f, **kwargs)
+                return f
+            wrapped()
```

Comment:
With the given change set, the decorator will return a None value. So, in this instance the following will happen
```python
@gendo.listen_for(rule='something')
def my_great_handler(user, message, **kwargs):
    # do something useful
   pass

>>> assert my_great_handler is None
# This statement is True and won't raise an AssertionError
```
The above code is equivalent to:

```python
def my_great_handler(user, message, **kwargs):
   # do something useful 
   pass

# Make a temp reference to the my_great_handler function object
tmp_function_reference = my_great_handler

# Now, we'll apply the decorator imperatively
decorator = gendo.listen_for(rule='something')
my_great_handler = decorator(my_great_handler)
```
Then the following will be true:
```python
>>> assert my_great_handler is None
# The above is still True since the decorator doesn't return anything.
# But the function object created above _will_ be in the gendo listeners list::
>>> all_view_funcs = [view_func for _, view_func, _ in gendo.listeners]
>>> assert tmp_function_reference in all_view_funcs  # this will be true.

# Attempting to call the decorated function will throw an exception
>>> my_great_handler(...)
TypeError: 'NoneType' object is not callable